parser/sql/dialect/mysql/src/main/antlr4/imports/mysql/Literals.g4

```diff
@@ -66,10 +66,15 @@ BIT_NUM_
     ;
 
 IDENTIFIER_
-    : [A-Za-z_$0-9\u0080-\uFFFF]*?[A-Za-z_$\u0080-\uFFFF]+?[A-Za-z_$0-9\u0080-\uFFFF]*
+    :  IP_ADDRESS
```

Comment:
Can you move ip address parse to BaseRule? You can refer mysql parse logic in sql_yacc.yy file.

```
ident_or_text:
          ident           { $$=$1;}
        | TEXT_STRING_sys { $$=$1;}
        | LEX_HOSTNAME { $$=$1;}
        ;

role_ident_or_text:
          role_ident
        | TEXT_STRING_sys
        | LEX_HOSTNAME
        ;

user_ident_or_text:
          ident_or_text
          {
            if (!($$= LEX_USER::alloc(YYTHD, &$1, NULL)))
              MYSQL_YYABORT;
          }
        | ident_or_text '@' ident_or_text
          {
            if (!($$= LEX_USER::alloc(YYTHD, &$1, &$3)))
              MYSQL_YYABORT;
          }
        ;
```

Reply:
Following is ShardingSphere BaseRule.g4 content:

```
textOrIdentifier
    : identifier | string_
    ;
    
variable
    : userVariable | systemVariable
    ;
    
userVariable
    : AT_ textOrIdentifier
    | textOrIdentifier
    ;
```

Reply:
Hi, I tried defining the syntax parsing rules for IP addresses in BaseRule.g4. 
However, during testing, I found that IP addresses were being matched by the NUMBER_ lexical rule in Literals.g4 file first. 
I have tried defining the rules multiple times, but the NUMBER_ rule always takes priority.

I have defined the IP_ADDRESS lexical rule in the Literals.g4 file and referenced it in the ipAddress rule in the BaseRule.g4 file. Would this be a correct approach?

Reply:
> I have defined the IP_ADDRESS lexical rule in the Literals.g4 file and referenced it in the ipAddress rule in the BaseRule.g4 file. Would this be a correct approach?

Yes, that's what I mean. Instead of changing the definition of the identifier, add an ipAddress rule to the textOrIdentifier.